src/main.rs

```diff
@@ -221,101 +229,100 @@ async fn process_created(path: PathBuf, uid: u32, gid: u32, workingdir: PathBuf)
 ///
 /// Should be none. Just early returns in case the path do not match.
 #[instrument]
-async fn process_removed(path: PathBuf, workingdir: PathBuf) {
-    let re = Regex::new("/dev/sd[a-zA-Z]{1,2}[0-9]{1,2}").unwrap();
-    if re.find(path.as_os_str().to_str().unwrap_or("")).is_none() {
-        return;
-    };
+async fn process_removed(path: PathBuf, workingdir: PathBuf) -> anyhow::Result<()> {
+  let re = Regex::new("/dev/sd[a-zA-Z]{1,2}[0-9]{1,2}").unwrap();
+  if re.find(path.as_os_str().to_str().unwrap_or("")).is_none() {
+    return Ok(());
+  };
 
-    guard!(let Some(device) = extract_device(path.as_path()).await else {
-      tracing::warn!("couldn't match the device for some reason?");
-      return;
-    });
-    tracing::info!("processing {:?}", path);
+  guard!(let Some(device) = extract_device(path.as_path()).await else {
+    tracing::warn!("couldn't match the device for some reason?");
+    return Ok(());
+  });
 
-    let dest: std::path::PathBuf = workingdir.join(PathBuf::from(device));
-    tracing::debug!("destpath is {:?}", dest);
+  tracing::info!("processing {:?}, event=removed", path);
+  notify(format!("removed device: {}", device).as_str())?;
 
-    if let Err(why) = unmount(&dest, UnmountFlags::FORCE) {
-        tracing::error!("failed to unmount {:?}: {}", path, why);
-    }
+  let dest: std::path::PathBuf = workingdir.join(PathBuf::from(device));
+  tracing::debug!("destpath is {:?}", dest);
+
+  if let Err(why) = unmount(&dest, UnmountFlags::FORCE) {
+    tracing::error!("failed to unmount {:?}: {}", path, why);
+  }
 
-    drop(remove_dir(dest).await);
+  drop(remove_dir(dest).await);
```

Comment:
Why an explicit drop here? You'll get that effect in the next line on returning the Ok

Reply:
fixed